toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -920,21 +940,111 @@ func createUserWithPassword(installChroot *safechroot.Chroot, user configuration
 		return
 	}
 
-	err = user.PasswordExpiresDaysIsValid()
-	if err != nil {
-		return
-	}
-
 	// Update password expiration
 	if user.PasswordExpiresDays != 0 {
+		// Ignore updating if there is no shadow file to update
+		if exists, _ := file.PathExists(shadowFile); !exists {
+			logger.Log.Debugf("No shadow file to update. Skipping.")
+			return
+		}
+
 		err = installChroot.UnsafeRun(func() error {
-			return shell.ExecuteLive(squashErrors, "chage", "-M", strconv.FormatInt(user.PasswordExpiresDays, passwordExpiresBase), user.Name)
+			return chage(fmt.Sprintf("%d", user.PasswordExpiresDays), user.Name)
 		})
 	}
 
 	return
 }
 
+// chage works in the same way as invoking "chage -M passwordExpirationInDays username"
+// i.e. it sets the maximum password expiration date.
+func chage(passwordExpirationInDays string, username string) (err error) {
+	var (
+		shadow             []string
+		passwordExpiration int64
+		usernameWithColon  = fmt.Sprintf("%s:", username)
+	)
+
+	shadow, err = file.ReadLines(shadowFile)
+	if err != nil {
+		return
+	}
+
+	passwordExpiration, err = strconv.ParseInt(passwordExpirationInDays, 10, 64)
+	if err != nil {
+		return
+	}
+
+	for n, entry := range shadow {
+		done := false
+		// Entries in shadow are separated by colon and start with a username
+		// Finding one that starts like that means we've found our entry
+		if strings.HasPrefix(entry, usernameWithColon) {
+			// Each line in shadow contains 9 fields separated by colon ("") in the following order:
+			// login name, encrypted password, date of last password change,
+			// minimum password age, maximum password age, password warning period,
+			// password inactivity period, account expiration date, reserved field for future use
+			const (
+				loginNameField         = 0
+				encryptedPasswordField = 1
+				passwordChangedField   = 2
+				minPasswordAgeField    = 3
+				maxPasswordAgeField    = 4
+				warnPeriodField        = 5
+				inactivityPeriodField  = 6
+				expirationField        = 7
+				reservedField          = 8
+			)
+
+			fields := strings.Split(entry, ":")
+			// Any value other than 9 indicates error in parsing
+			if len(fields) != 9 {
```

Comment:
nit: Change `9` here to be a const defined above, such as `expectedNumberOfFields`.